packages/components/src/settings-form/setting-types/setting-select.tsx

```diff
@@ -0,0 +1,51 @@
+/**
+ * External dependencies
+ */
+import { useMemo } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { SelectControl } from '../../index';
+import { ControlProps, StringToString } from '../types';
+
+interface SelectControlOption {
+	key: string;
+	label: string;
+	value: { id: string };
+}
+
+const transformOptions = ( options: StringToString ) => {
+	return Object.keys( options ).reduce(
+		( all: SelectControlOption[], curr ) => {
+			all.push( {
+				key: curr,
+				label: options[ curr ],
```

Comment:
This function is indexing `options` within the reduce, and returning an array, which makes the reduce redundant.  This makes me think `Object.entries` with `map` would be a lot simpler here.

e.g. 

```
return Object.entries( options ).map(([key,value]) => { key: key, label: value, value: { id: key }});
```

Reply:
Totally true. It seems I'm a reduce addict. 💊